arbor/backends/gpu/shared_state.cpp

```diff
@@ -283,14 +292,22 @@ void shared_state::update_prng_state(mechanism& m) {
         // The RNG will be seeded by a global seed, the mechanism id, the variable index, the
         // current site's global cell, the site index within its cell and a counter representing
         // time.
-        generate_normal_random_values(
-            m.ppack_.width,          // number of sites
-            cbprng_seed,             // simulation seed value
-            mech_id,                 // mechanism id
-            counter,                 // step counter
-            store.prng_indices_d_,   // additional indices (gid and per-cell counter)
-            store.random_numbers_d_  // destination
-        ); 
+        const auto num_rv = store.random_numbers_d_[0].size();
+        const auto width_padded = store.value_width_padded;
+        const auto width = m.ppack_.width;
+#ifdef ARB_ARBOR_NO_GPU_RAND
```

Comment:
This is my only painpoint left:
Could we make it so that shared state doesn't need to know about this flag? Currently it's used in ~three~
four different places, which might cause issues down the road. How about delegating these things to the
pRNG impl
- writing the values (here)
- setting gid/idx (ll 358)
- allocating (ll 396)/ filling (ll 477) the indices

via the use of an appropriate interface. In said interface we can make the decision _once_.